### ISSUE TITLE
Option to load icons from websites OR at least assign a custom icon file instead

### ISSUE THREAD

**Issue by nickbe** [buttercup/buttercup-desktop#985]
Again I would like to ask for this option. If you don't want to get icons from websites then
it would be great to have at least a simple option to import icons from local png or ico files.

That way we can at least have a manual way to have custom icons.
Would that be an option?

**Comment by perry-mitchell** [comment]
I'm happy to plan alternatives to the icon fetcher we host. Please don't take anything I ask here as a _no_ for this issue - but what do you suppose will be the adoption rate for such a feature? I for one wouldn't want to manage an icon file _just_ for the UI of my password manager.

Can we consider some alternatives as well? What if we allowed users to self-host this icon proxy (being built into the desktop app)? 

The most popular icon-based options would be the following, in my mind:
 * Auto (as is now)
 * Off (no fetching ever)
 * Local (proxy option hosted internally in the app, as mentioned above)

These options are easier as users don't need to think about adding/managing icons. It either works or it doesn't :)

**Comment by nickbe** [comment]
Well problem is that 90% of all icons in our collections are not the usual social hype platform stuff, but completely custom sites, platforms and software. So all icons look the same here except for a few.

"Hosting the proxy in the app" - well I have no idea how this would work. Would this enable me to tell the proxy to again get the icons directly from the sites?

**Comment by kolole** [comment]
Buttercup-desktop for Linux have no options to add icons, either manually or automatically, as far as I can see. I'm using the latest (2.5.0) Appimage.

**Comment by skillz-xx** [comment]
to me, when realizing that, it was a HUGE downer.. and I cannot find an option to turn the icon checking in general OFF in the application. I dont want it, I dont need it, but now all the sites I configure in the application make contact with my ip? that is a very bad approach and a huge privacy concern for me. 
In Ubuntu desktop version I cannot turn this off? seriously?

please let me know!

**Comment by skillz-xx** [comment]
for example the service doesnt find icons for .onion sites
and i dont want the app to send such requests anyway

**Comment by perry-mitchell** [comment]
> now all the sites I configure in the application make contact with my ip

No, all icons are loaded via our proxy that tracks **nothing**.

> In Ubuntu desktop version I cannot turn this off? seriously?

You're welcome to submit a PR with this feature, I would accept it.
